Fix Lex and Yac compilation CMake config errors on Windows

marcboulle opened the pull request:
Correction:
- fichier src\Learning\KWData\CMakeLists.txt : 'COMPILE_FLAGS-- nounistd' -> 'COMPILE_FLAGS "--nounistd"'
- le fichier KWCYac.hpp, non utilise, est genere par bison: ajout dans le .gitignore

Procedure d'installation (documentee dans le wiki):
- installation de la derniere version de windows flex et bison
  - derniere version: version 2.5.25 (bison 3.8.2, flex 2.6.4) https://github.com/lexxmark/winflexbison/releases (cela marche avec les versions 2.5.24 et 2.5.25)
- mettre le repertoire des binaires win_flex.exe et win_bison.exe dans le path

Cohabitation avec 'Gow - The lightweight alternative to Cygwin': https://github.com/bmatzelle/gow
- si gow est installe, cmake va chercher les binaires de flex et yac dans le repertoire bin de gow, et cela ne marche pas
- contournement pour garder gow sur sa machine:
  - dans le repertoire bin de gow, renommer flex.exe en NO_flex.exe et bison.exe en NO_bison.exe

 Mise en oeuvre, le temps d'une compilation lex et yac
 - dans le fichier CMakePreset.json, mettre "BUILD_LEX_YACC": "ON"
 - attention à "Supprimer le cache et configurer" dans le menu "Project" de Visual Studio
   - parfois, un peu laborieux, et necessaite de recommencer plusieurs fois

marcboulle committed:
Fix install action yml to comply with new LearningTest scripts

Correction:
  - .github\actions\test-khiops-install\action.yml: correction pour mise en phase avec les nouveaux scripts de LearningTest
  - test\LearningTestTool\py\kht_apply.py: concatenation des dictionnaire dans register_all_instructions,
    sans utiliser l'operateur '|' supporte uniquement depuis python 3.9

bruno-at-orange reviewed (approved):
LGTM

folmos-at-orange reviewed (approved):
LGTM

Next time separe non related fixes into two commits

marcboulle committed:
Fix Lex and Yac compilation CMake config errors on Windows

Correction:
- fichier src\Learning\KWData\CMakeLists.txt : 'COMPILE_FLAGS-- nounistd' -> 'COMPILE_FLAGS "--nounistd"'
- un fichier KWCYac.hpp est maintenant genere par bison, et ajoute au projet dans KWData

Procedure d'installation, documentee dans le wiki 'Setting Up the Development Environment'
- installation de la derniere version de windows flex et bison
  - derniere version: version 2.5.25 (bison 3.8.2, flex 2.6.4)
    https://github.com/lexxmark/winflexbison/releases
    (cela marche avec les versions 2.5.24 et 2.5.25)
- mettre le repertoire des binaires win_flex.exe et win_bison.exe dans le path

Cohabitation avec 'Gow - The lightweight alternative to Cygwin': https://github.com/bmatzelle/gow
- si gow est installe, cmake va chercher les binaires de flex et yac dans le repertoire bin de gow, et cela ne marche pas
- contournement pour garder gow sur sa machine:
  - dans le repertoire bin de gow, renommer flex.exe en NO_flex.exe et bison.exe en NO_bison.exe

 Mise en oeuvre, le temps d'une compilation lex et yac
 - dans le fichier CMakePreset.json, mettre "BUILD_LEX_YACC": "ON"
 - attention à "Supprimer le cache et configurer" dans le menu "Project" de Visual Studio
   - parfois, un peu laborieux, et necessaite de recommencer plusieurs fois
